eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelay.scala

```diff
@@ -199,10 +201,43 @@ class NodeRelay private(nodeParams: NodeParams,
             context.log.warn(s"rejecting trampoline payment reason=$failure")
             rejectPayment(upstream, Some(failure))
             stopping()
+          case None =>
+            if (nextPayload.isAsyncPayment && nodeParams.features.hasFeature(Features.AsyncPaymentPrototype)) {
+              waitForTrigger(upstream, nextPayload, nextPacket)
+            } else {
+              doSend(upstream, nextPayload, nextPacket)
+            }
+        }
+      case RelayAsyncPayment => context.log.debug("received async payment trigger while waiting to receive all incoming payment packets")
+        receiving(htlcs, nextPayload, nextPacket, handler)
```

Comment:
nit:
```suggestion
      case RelayAsyncPayment =>
        context.log.debug("received async payment trigger while waiting to receive all incoming payment packets")
        receiving(htlcs, nextPayload, nextPacket, handler)
```

Reply:
Actually, I think we can remove this case entirely. As you pointed out earlier, the trigger won't be returned until we register for it in the `waitForTrigger` state. Fixed in  [6400ce2](https://github.com/ACINQ/eclair/pull/2435/commits/6400ce2070e4aab21fc08801c1afe44aaa2d7747)